### ISSUE TITLE
Hi, after the update it seems to complain about ch_mult's default value ''train_vae.py: error: argument --ch_mult: invalid int value: '1,1,2,2,4,''

### ISSUE THREAD

**Issue by jordanvalter** [tgisaturday/dalle-lightning#17]
WARNING:root:TPU has started up successfully with version pytorch-1.9
GPU available: False, used: False
TPU available: True, using: 8 TPU cores
IPU available: False, using: 0 IPUs
usage: train_vae.py [-h] [--train_dir TRAIN_DIR] [--val_dir VAL_DIR]
                    [--log_dir LOG_DIR] [--backup_dir BACKUP_DIR]
                    [--ckpt_path CKPT_PATH] [--resume] [--backup]
                    [--backup_steps BACKUP_STEPS] [--log_images]
                    [--image_log_steps IMAGE_LOG_STEPS]
                    [--refresh_rate REFRESH_RATE] [--precision PRECISION]
                    [--fake_data] [--use_tpus] [--seed SEED] [--gpus GPUS]
                    [--gpu_dist] [--tpus TPUS]
                    [--num_sanity_val_steps NUM_SANITY_VAL_STEPS]
                    [--learning_rate LEARNING_RATE] [--lr_decay]
                    [--starting_temp STARTING_TEMP] [--temp_min TEMP_MIN]
                    [--anneal_rate ANNEAL_RATE] [--batch_size BATCH_SIZE]
                    [--epochs EPOCHS] [--num_workers NUM_WORKERS]
                    [--img_size IMG_SIZE] [--resize_ratio RESIZE_RATIO]
                    [--test] [--debug] [--xla_stat] [--web_dataset]
                    [--dataset_size DATASET_SIZE] [--model MODEL]
                    [--codebook_dim CODEBOOK_DIM] [--num_tokens NUM_TOKENS]
                    [--double_z DOUBLE_Z] [--z_channels Z_CHANNELS]
                    [--resolution RESOLUTION] [--in_channels IN_CHANNELS]
                    [--out_channels OUT_CHANNELS] [--hidden_dim HIDDEN_DIM]
                    [--ch_mult CH_MULT [CH_MULT ...]]
                    [--num_res_blocks NUM_RES_BLOCKS]
                    [--attn_resolutions ATTN_RESOLUTIONS [ATTN_RESOLUTIONS ...]]
                    [--dropout DROPOUT] [--quant_beta QUANT_BETA]
                    [--quant_ema_decay QUANT_EMA_DECAY]
                    [--quant_ema_eps QUANT_EMA_EPS] [--num_res_ch NUM_RES_CH]
                    [--smooth_l1_loss] [--kl_loss_weight KL_LOSS_WEIGHT]
                    [--disc_conditional DISC_CONDITIONAL]
                    [--disc_in_channels DISC_IN_CHANNELS]
                    [--disc_start DISC_START] [--disc_weight DISC_WEIGHT]
                    [--codebook_weight CODEBOOK_WEIGHT] [--wandb]
train_vae.py: error: argument --ch_mult: invalid int value: '1,1,2,2,4,'

**Comment by tgisaturday** [comment]
@jordanvalter Can you give me more specific situation like actual command you are using?

**Comment by tgisaturday** [comment]
@jordanvalter I've just tested on Google cloud TPUs and it works. Wonder if you are using colab since I didn't update the colab notebook yet. Try using config files like https://github.com/tgisaturday/dalle-lightning/blob/master/configs/vqvae_f8.yaml

**Comment by jordanvalter** [comment]
Will do ill close and reopen if issue persists

**Comment by js12312** [comment]
> @jordanvalter I've just tested on Google cloud TPUs and it works. Wonder if you are using colab since I didn't update the colab notebook yet. Try using config files like https://github.com/tgisaturday/dalle-lightning/blob/master/configs/vqvae_f8.yaml

Thank you so much that worked!

**Comment by JiaLim98** [comment]
Hi @tgisaturday,

Thank you so much for the great work!

I reproduced the same issue mentioned above by @jordanvalter by simply running each cell in the Colab file.
![image](https://user-images.githubusercontent.com/66768413/129851626-c259a5cc-eb6d-4831-9913-7e5b4d1b3543.png)
I am curious about whether an 'int' argument can accept a list of integers.
![image](https://user-images.githubusercontent.com/66768413/129851838-d7ef2588-715f-4764-a1e4-8c4f4288bb6a.png)
It would be great if you could help me with this issue. Thanks in advance!

Regards,
JiaLim

**Comment by tgisaturday** [comment]
@JiaLim98 @jordanvalter I'm not quite used to colab. Could you try --ch_mult 1 1 2 2 4 ?

**Comment by JiaLim98** [comment]
@tgisaturday No, it does not work, because we are now passing 5 integers into a argument, which is wrong.
Do you have any alternative commands/tutorials that might get his repo working?

**Comment by tgisaturday** [comment]
@JiaLim98 Try using configs. You can find some examples in configs directory.

**Comment by JiaLim98** [comment]
Hi @tgisaturday, I have got this repo working! Thanks again!
I have some questions to ask:
At what stage will you recommend to stop the training of vae and start training dalle?
Another question is that, how do I run inference (i.e. generate images from text) using trained dalle?

**Comment by tgisaturday** [comment]
@JiaLim98 Great! Please share how you solved the problems if you don't mind. The training schedule depends on your data but 20 epochs of training vae and start training dalle would be enough. To check the image generation quality you can use --log_images but might slowdown dalle training if you use tpu (gpus are fine). You may also use generate_images func in dalle model.

**Comment by JiaLim98** [comment]
@tgisaturday Sorry for the late reply.

I think the problem is caused by variable parsing using the script file 'run.sh'. I bypassed the two cells (creating the script file and running the script command `!(sh /content/tmp/run.sh)` and called train_vae.py directly using `!python /content/lib/dalle-lightning/train_vae.py --gpus 1 --train_dir /content/data/train --val_dir /content/data/val --batch 1`. The parameters are set for test run on Colab only, anyone who runs locally can adjust accordingly.

Nevertheless, thank you again for this amazing work and thank you for the information given!
